src/Containers/ReactTable.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useTable, Column, TableHeaderProps, TableRowProps } from 'react-table';
+import { IProfileProps } from './VendorsList/Profile';
+import { MainInterface, ResponsiveInterface } from '../Utils/BaseStyles';
+import { media, scroll } from '../Utils/Mixins';
+
+export interface IVendorsData extends IProfileProps {
+    tags?: string[];
+    createdAt?: string;
+};
+
+export interface IReactTableProps extends MainInterface, ResponsiveInterface,React.HTMLAttributes<HTMLDivElement> {
+    data: IVendorsData[];
```

Comment:
This component should be fully reusable for any component besides the typing. Is there any way to pass in a general interface here? or is that the type any?

Reply:
You could do object[]
But the best thing to do is most likely involving typescript generics :)